### PR TITLE
Add support for PRNG on intel_gen

### PR DESCRIPTION
**Pull request by YangleiZouIntel**
* Use the x86 implementation of PRNG to execute the op on the host.
* Lots of buildifier formatting changes (please remember to use buildifier for BUILD files).

### CODE REVIEW

File: pmlc/target/intel_gen/pipeline.cc [hunk 1]
```diff
@@ -35,8 +35,30 @@ using namespace mlir; // NOLINT[build/namespaces]
...
+    conversion::pxa_to_affine::PXAToAffineConversionTarget target(ctx);
+
+    OwningRewritePatternList patterns;
+    populatePXAPrngToAffineConversionPatterns(patterns, &ctx);
+    conversion::pxa_to_affine::populatePXAToAffineConversionPatterns(patterns,
+                                                                     &ctx);
+
+    if (failed(applyPartialConversion(getOperation(), target, patterns,
+                                      nullptr))) {
+      getOperation().dump();
```
Comment (flaub) [inline pmlc/target/intel_gen/pipeline.cc:55]: Drop this.

File: pmlc/target/intel_gen/pipeline.cc [hunk 2]
```diff
@@ -35,8 +35,30 @@ using namespace mlir; // NOLINT[build/namespaces]
...
+      emitError(UnknownLoc::get(&ctx), "Error lowering pxa -> affine\n");
```
Comment (flaub) [inline pmlc/target/intel_gen/pipeline.cc:56]: Use `getOperation()->emitError()`

File: pmlc/target/intel_gen/passes.td [hunk 3]
```diff
@@ -28,6 +28,13 @@ def ConvertParallelLoopToGpu : Pass<"intel-gen-convert-parallel-loops-to-gpu"> {
   let dependentDialects = ["mlir::AffineDialect", "mlir::gpu::GPUDialect"];
 }
 
+def ConvertPXAToAffine : Pass<"intel_gen-convert-pxa-to-affine", "mlir::ModuleOp"> {
```
Comment (flaub) [inline pmlc/target/intel_gen/passes.td:31]: Use `intel-gen` instead of `intel_gen`

**Comment by flaub** [comment]
This looks identical to the x86 target implementation. IIUC, the strategy here is to implement the PRNG on the host-side. If so, then I think we should use the x86 implementation directly, instead of making a copy.

**Comment by YangleiZouIntel** [comment]
> This looks identical to the x86 target implementation. IIUC, the strategy here is to implement the PRNG on the host-side. If so, then I think we should use the x86 implementation directly, instead of making a copy.

Thanks. Yes it is to implement PRNG on the host side for now to run resnet50 on gpu. Do you think it is better to move prng lowering under pmlc/target/x86 to pmlc/conversion/pxa_to_affine/ ? Otherwise I think we will need to add x86 as a dependent to intel_gen which seems not good.

File: pmlc/target/intel_gen/passes.td (hunk 3 above)
Comment (YangleiZouIntel) [reply pmlc/target/intel_gen/passes.td]: Fixed

File: pmlc/target/intel_gen/pipeline.cc (hunk 2 above)
Comment (YangleiZouIntel) [reply pmlc/target/intel_gen/pipeline.cc]: Fixed

**Comment by flaub** [comment]
Yes, I think moving it to `pxa_to_affine` makes some sense. Or we could leave in `x86` but just make a library out of this one pattern so it can be reused by `intel_gen`. It's interesting to think about whether we should have `host` and `device` targets as a further distinction.

**Comment by adstraw** [comment]
Are we thinking about this as a permanent fixture of the GPU pipeline or a temporary workaround until we figure out PRNG lowering on the GPU?  My understanding was that it was temporary.  If so, perhaps it is ok to let this through without further design review.  If permanent, or if we think we need host side compute for other things in the GPU pipeline then it's a different story.

**Comment by flaub** [comment]
I'll fix this in my upcoming PR, stay tuned :)

**Comment by YangleiZouIntel** [comment]
> I'll fix this in my upcoming PR, stay tuned :)

Thank you. I was running out of time yesterday. Looking forward to seeing the new PR.

**Review by flaub — APPROVED** [review]

**Comment by adstraw** [comment]
Nice work @flaub !

**Pull request merged**